rsocket-core/src/main/java/io/rsocket/fragmentation/FragmentationUtils.java

```diff
@@ -0,0 +1,34 @@
+package io.rsocket.fragmentation;
+
+import io.netty.buffer.ByteBuf;
+import io.rsocket.Payload;
+import io.rsocket.frame.FrameHeaderFlyweight;
+import io.rsocket.frame.FrameLengthFlyweight;
+
+public final class FragmentationUtils {
+  public static boolean isValid(int mtu, Payload payload) {
+    return payload.hasMetadata()
+        ? isValid(mtu, payload.data(), payload.metadata())
+        : isValid(mtu, payload.metadata());
```

Comment:
/!\ passing `metadata()` to `isValid(mtu, data)`
(a unit test should probably be added)

Reply:
Yeah. It is a bag. I was sure to test is covering that. Will improve

Reply:
fixed